Correctly create dde network on system:update

renestalder opened the pull request:
refs #70

This PR addresses issue https://github.com/whatwedo/dde/issues/70 by ensuring the Docker network is properly created during system updates.

Key changes:

- Extracted network creation logic into a dedicated _ensureNetwork internal function
- ~~Added network creation to system:update workflow after system destruction~~
- ~~Refactored system:up to use the new shared function, eliminating code duplication~~
- Made the network filters to search for exact matches of `dde` instead of substring matches to resolve issues with other networks containing `dde` in their naming somewhere.

renestalder committed:
refactor(commands.system.up): externalize network check and creation

Move the network existence check and creation to an internal function to
allow reusing it in other commands.

Signed-off-by: René Stalder <rene@whatwedo.ch>

s4mpl3d commented on commands/system/update.sh:
```diff
@@ -8,6 +8,8 @@ function system:update() {
     _logRed "Removing dde (system)"
     system:destroy
 
+    _ensureNetwork
```
Seems like it. There are some mentions of this in the forums: https://forums.docker.com/t/error-starting-userland-proxy-listen-tcp4-0-0-0-0-bind-address-already-in-use/140275/14 . Only issue i could find on gh is closed https://github.com/docker/compose/issues/10846 and somehow i cant reproduce this behavior. 
I assume you also have a current docker version? Maybe it's a macos thing

renestalder replied:
@robinlehrmann Nope, doesn't make a difference in my case.

Though in proceeding going down the whole code logic and running it bit by bit I found a crucial issue in the network check, and therefore removal and creation logic:

```
docker network ls --filter=name=dde -q
```

This is not an exact filter. This is a substring filter. This also shows any other networks that have some form of wording with `dde` in it. So this is also reporting for like `ddev` networks. So that whole routine here fails as soon as other networks have `dde` somewhere in their name. And the solution is as simple as that:

```
docker network ls --filter=name=^dde$ -q
```

So might have been a very lucky shot the additional `_ensureNetwork` call this solve the problem in the first testings. I might have cleaned up all my docker environments in the process that it worked.

@robinlehrmann @s4mpl3d I'm gonna remove the redundant `_ensureNetwork` and update the filter routines.

renestalder committed:
fix: filter for exact network match named `dde` instead of substring

Resolve an issue where network destroyal and creation checks would yield
false positives when other networks containing `dde` in their name would
be present, making these routines fail half way through.

refs: #70
Signed-off-by: René Stalder <rene@whatwedo.ch>
